pkg/virt-controller/watch/vmi.go

```diff
@@ -2399,3 +2401,97 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+type aggregateDataVolumeCondition struct {
+	Status  k8sv1.ConditionStatus
+	Reason  []string
+	Message []string
+}
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
```

Comment:
```suggestion
func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
	dvsReadyCondition := virtv1.VirtualMachineInstanceCondition{
		Status: k8sv1.ConditionTrue,
		Type:   virtv1.VirtualMachineInstanceDataVolumesReady,
		Reason: virtv1.VirtualMachineInstanceReasonAllDVsReady,
		Message: "All of the VMI's DVs are bound and not running",
	}

	for _, dv := range dvs {
		if dv.Status.Phase != cdiv1.Succeeded { 
			dvsReadyCondition.Message = "Not all of the VMI's DVs are ready"
			dvsReadyCondition.Status = k8sv1.ConditionFalse
			break
		}
	}

	vmiConditions := controller.NewVirtualMachineInstanceConditionManager()
	vmiConditions.UpdateCondition(vmiCopy, &dvsReadyCondition)
}
```

Reply:
> 		if dv.Status.Phase != cdiv1.Succeeded { 

According to [K8s API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties), we should explicitly monitor the individual conditions and do not implicitly infer from phases.

See:

> Some resources in the v1 API contain fields called phase, and associated message, reason, and other status fields. **The pattern of using phase is deprecated. Newer API types should use conditions instead.** Phase was essentially a state-machine enumeration field, that contradicted [system-design principles](https://git.k8s.io/design-proposals-archive/architecture/principles.md#control-logic) and hampered evolution, since [adding new enum values breaks backward compatibility](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md). Rather than encouraging clients to infer implicit properties from phases, we prefer to explicitly expose the individual conditions that clients need to monitor. Conditions also have the benefit that it is possible to create some conditions with uniform meaning across all resource types, while still exposing others that are unique to specific resource types. See [#7856](http://issues.k8s.io/7856) for more details and discussion.

Reply:
I'm okay with monitoring the conditions only, but I think we should try to fail fast and early and try not to replace the info that CDI already exposes. This would simplify the code a lot.

Reply:
I pushed a simplified version (looking only at cdiv1.DataVolumeReady conditions) at https://github.com/kubevirt/kubevirt/pull/10905